server/game/cards/02_SHD/units/FinalizerMightOfTheFirstOrder.ts

```diff
@@ -0,0 +1,62 @@
+import AbilityHelper from '../../../AbilityHelper';
+import * as Helpers from '../../../core/utils/Helpers.js';
+import { Card } from '../../../core/card/Card';
+import { TriggeredAbilityContext } from '../../../core/ability/TriggeredAbilityContext';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { RelativePlayer, TargetMode, WildcardCardType, WildcardZoneName, ZoneName } from '../../../core/Constants';
+
+export default class FinalizerMightOfTheFirstOrder extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '9752523457',
+            internalName: 'finalizer#might-of-the-first-order',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addWhenPlayedAbility({
+            title: 'Choose any number of friendly units',
+            targetResolver: {
+                activePromptTitle: 'Choose friendly units that will capture enemy units in the same arena',
+                mode: TargetMode.UpToVariable,
+                cardTypeFilter: WildcardCardType.Unit,
+                zoneFilter: WildcardZoneName.AnyArena,
+                controller: RelativePlayer.Self,
+                canChooseNoCards: true,
+                numCardsFunc: (context) => Math.min(
+                    context.source.controller.getUnitsInPlay().length,
+                    this.countOpponentNonLeaderUnitsInPlay(context, WildcardZoneName.AnyArena)
+                ),
+                multiSelectCardCondition: (card, selectedCards, context) => this.countOpponentNonLeaderUnitsInPlay(context, card.zoneName) > this.countSelectedCardsInZone(selectedCards, card.zoneName),
```

Comment:
Is this too much complexity? I guess it would work fine using `TargetMode.Unlimited` too.

Reply:
Nah I think this is fine, this is a weird card so it's okay if it ends being a bit complicated. I doubt there'll be another ability like this one anytime soon